### ISSUE TITLE
@onlyif with some constants

### ISSUE THREAD

**Issue by MatisPatel** [JuliaDynamics/DrWatson.jl#250]
**Is your feature request related to a problem? Please describe.**
I have a parameter B that greatly increases search space. I want to investigate b only when some values of A are hit but I still want as a base line all values of A to be run on a subset of B. 

**Describe the solution you'd like**
```julia
d = Dict(:a => [1,2,3], :b => [1,2, @onlyif(:a==2, collect(3:5))])
```
In the above line @onlyif doesn't expand across the container `collect(3:5)`. It just makes one dictionary with :b of that dict being an iterator `3:5`.

**Describe alternatives you've considered**
```julia
d = Dict(:a => [1,2,3], :b => [1,2, @onlyif(:a==2, collect(3:5))])
listd = dict_list.(dict_list(d))
flatList = collect(Iterators.flatten(listd))
```
The above does work but it a tad messy. Maybe this is just a reasonable issue and the workaround is just the way to do it.

**Comment by JonasIsensee** [comment]
```julia
d = Dict(:a => [1,2,3], :b => [1,2, @onlyif(:a==2, collect(3:5))...])
```

Splatting the `@onlyif` should work.

**Comment by MatisPatel** [comment]
Yep that works! Obvious in retrospect. Maybe this use case could just be added as an example to the docs?

**Comment by Datseris** [comment]
Please do a PR that adds it to the "real world examples"?
